src/Catalog/Dnx/DnxMaker.cs

```diff
@@ -91,6 +116,16 @@ public DnxMaker(StorageFactory storageFactory)
             var destinationStorage = _storageFactory.Create(packageId);
             var nuspecUri = await SaveNuspecAsync(destinationStorage, packageId, normalizedPackageVersion, nuspec, cancellationToken);
             var nupkgUri = await CopyNupkgAsync(sourceStorage, destinationStorage, packageId, normalizedPackageVersion, cancellationToken);
+            if (!string.IsNullOrWhiteSpace(iconFilename))
```

Comment:
This should be done before .nupkg copy. In `DnxCatalogCollector` the .nupkg is used as a marker that this code has finished running.